qiskit/mapper/mapping.py

```diff
@@ -60,8 +60,10 @@ def layer_permutation(layer_partition, layout, qubit_subset, coupling, trials,
     swap circuit has been applied. The trivial_flag is set if the layer
     has no multi-qubit gates.
     """
-    if seed is not None:
-        np.random.seed(seed)
+    if seed is None:
+        seed = np.random.randint(0, np.iinfo(np.int32).max)
```

Comment:
If `seed` is `None`, `RandomState` should be "initialized" as `np.random.RandomState(None)`. Also noticed that 
`np.iinfo(np.int32).max` (2147483647)` ≠ `2**32 - 1` (4294967295).

Reply:
- The whole point of this PR is to have accessible seed values.  If `None` is passed to `RandomState` then the seed is picked from the clock time in c++, and it is inaccessible to Python.  So here we seed `RandomState` with a random int that we can get our hands on and log.

- Yes, I did int32 instead of uint32.  It is now fixed.

Reply:
Why not just `np.random.randint(0, 2**32 - 1)`?
```
In : timeit.timeit('np.iinfo(np.uint32).max', number=10000, setup='import numpy as np')                                                                                                                  
Out: 0.015713413013145328

In : timeit.timeit('2**32 - 1', number=10000)                                                                                                                               
Out: 0.0001322640455327928
```